[PowerToys] Add Command Not Found docs

stefansjfw opened the pull request:
To be merged only with the PowerToys 0.77 release

stefansjfw committed:
[PowerToys] Add Command Not Found docs

alvinashcraft commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,36 @@
+![AlwaysOnTop screenshot.](../images/pt-cmd-not-found.png)
+
+## Requirements
+ - [PowerShell 7](https://learn.microsoft.com/en-us/PowerShell/scripting/install/installing-PowerShell-on-windows?view=PowerShell-7.4)
```
```suggestion
 - [PowerShell 7](/PowerShell/scripting/install/installing-PowerShell-on-windows?view=PowerShell-7.4)
```

Is the `?view=PowerShell-7.4` query param needed, or can we let it default to the latest PowerShell docs?

Jay-o-Way replied:
```suggestion
 - [PowerShell 7](https://learn.microsoft.com/PowerShell/scripting/install/installing-PowerShell-on-windows?view=PowerShell-7.4)
```

alvinashcraft commented on hub/powertoys/cmd-not-found.md:
````diff
@@ -0,0 +1,36 @@
+#34de4b3d-13a8-4540-b76d-b9e8d3851756
+```
+
+Note: If there is no profile file at the time of installing the module, profile file will be created. After installing the module, restart PowerShell session to use the module.
````
```suggestion
**Note:** If there is no profile file at the time of installing the module, profile file will be created. After installing the module, restart PowerShell session to use the module.
```
Is bold sufficient, or do you think it should be formatted as a `[!NOTE]`?

Jay-o-Way replied:
Honestly and personally, this actually feels more like a normal paragraph to me.

stefansjfw replied:
Bold is ok. I'd go with bold to outline this, especially "restart PowerShell" part

alvinashcraft commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,36 @@
+
+## Install the module
+
+To install the Command Not Found module, navigate to PowerToys settings Command Not Found page and press Install button. On module install, PowerShell 7 experimental features needed for the module to function will be enabled:
```
```suggestion
To install the Command Not Found module, navigate to the Command Not Found page in PowerToys settings and select the **Install** button. Once the module installation has completed, PowerShell 7 experimental features needed for the module to function will be enabled:

```

Jay-o-Way commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,36 @@
+
+## Install the module
+
+To install the Command Not Found module, navigate to PowerToys settings Command Not Found page and press Install button. On module install, PowerShell 7 experimental features needed for the module to function will be enabled:
```
```suggestion
To install the Command Not Found module, go to PowerToys settings Command Not Found page and select **Install**. After installation, PowerShell 7 experimental features needed for the module to function will be enabled:
```

alvinashcraft replied:
I just left a suggestion for this same block. Can you combine the suggestions into an updated paragraph, @stefansjfw?

Jay-o-Way replied:
Haha, we're thinking the same thing! :)

alvinashcraft commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,36 @@
+
+## Uninstall the module
+
+To uninstall the Command Not Found module, navigate to PowerToys settings Command Not Found page and press Uninstall button. On module uninstall, previously added block of commands will be removed from PowerShell profile file. 
```
```suggestion
To uninstall the Command Not Found module, navigate to the Command Not Found page in PowerToys settings and select the **Uninstall** button. Once the module uninstall has completed, the block of commands previously added will be removed from the PowerShell profile file. 
```

Jay-o-Way replied:
```suggestion
To uninstall the Command Not Found module, go to PowerToys settings Command Not Found page and select **Uninstall**. After uninstallation, previously added block of commands will be removed from PowerShell profile file. 
```

Jay-o-Way commented on hub/powertoys/cmd-not-found.md:
````diff
@@ -0,0 +1,36 @@
+#34de4b3d-13a8-4540-b76d-b9e8d3851756
+```
+
+Note: If there is no profile file at the time of installing the module, profile file will be created. After installing the module, restart PowerShell session to use the module.
````
```suggestion
Note: The profile file will be created if needed. Restart PowerShell session to use the module.
```

stefansjfw committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Jay <65828559+Jay-o-Way@users.noreply.github.com>

stefansjfw committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Alvin Ashcraft <73072+alvinashcraft@users.noreply.github.com>

stefansjfw committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Alvin Ashcraft <73072+alvinashcraft@users.noreply.github.com>

stefansjfw committed:
Address PR comments

stefansjfw committed:
Fix copy-pasta

stefansjfw commented:
I believe I addressed all the comments. Thanks for the suggestions!

alvinashcraft reviewed (approved):
This looks good to me. We'll see if @Jay-o-Way has any other suggestions. Otherwise, I'll merge when the release goes out.

Jay-o-Way commented:
Looks good to me :)

NathanD-GDS commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,37 @@
+
+# Command Not Found utility
+
+A PowerShell 7 module that detects an error thrown by a command and suggests a relevant WinGet package to install, if available.
```
```suggestion
A PowerShell 7 module that detects command-line errors and suggests a relevant WinGet package to install, if available.
```

This is just a small suggestion but I'm not sure about 'thrown' as a verb to describe errors. I know it's used for [exceptions in these docs](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/exception-throwing) but I can't see it in many other Microsoft docs.

Jay-o-Way commented on hub/powertoys/cmd-not-found.md:
````diff
@@ -0,0 +1,37 @@
+ - PSCommandNotFoundSuggestion
+
+After that, PowerShell profile file will be appended with following block of PowerShell commands:
+```
````
What's the language for this code snippet? psh?

stefansjfw replied:
yes

Jay-o-Way replied:
```suggestion

```psh
```

Jay-o-Way commented on hub/powertoys/cmd-not-found.md:
```diff
@@ -0,0 +1,37 @@
+
+![Command Not Found screenshot.](../images/pt-cmd-not-found.png)
+
+## Requirements
```
```suggestion
## Requirements

```

Jay-o-Way reviewed (commented):
Nit bits

stefansjfw committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Jay <65828559+Jay-o-Way@users.noreply.github.com>

stefansjfw committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Nathan Driver <80888259+NathanD-GDS@users.noreply.github.com>

alvinashcraft committed:
Update cmd-not-found to add important note

Add note about PowerShell issues with some installers, linking to github issue in PT repo.

alvinashcraft committed:
Update hub/powertoys/cmd-not-found.md

Co-authored-by: Jay <65828559+Jay-o-Way@users.noreply.github.com>
